src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4064,9 +4064,12 @@ void ClassFileParser::check_super_class_access(const InstanceKlass* this_klass,
       return;
     }
 
-    if (super_ik->is_sealed() && !super_ik->has_as_permitted_subclass(this_klass)) {
-      classfile_icce_error("class %s cannot inherit from sealed class %s", super_ik, THREAD);
-      return;
+    if (super_ik->is_sealed()) {
+      stringStream ss;
```

Comment:
Seems like the ResourceMark belongs here above the has_as_permitted_subclass call as the as_C_string() functions are called in there.  Not in the classfile_icce_error.

Reply:
Ah yes, in this case that is true. Fixed.